README_fr.md

```diff
@@ -0,0 +1,1641 @@
+
+### Je veux lister les commits ayant des fichiers spécifiques
+
+Pour trouver tous les commits contenant un fichier spécifique, vous pouvez utiliser :
+
+```sh
+$ git log -- <chemin du fichier>
+```
+
+En général, vous préciseriez un chemin exact, mais vous pouvez aussi utiliser des wildcards dans le chemin et le nom du fichier :
+
+```sh
+$ git log -- **/*.js
+```
+
+En utilisant des wildcards, il est utile de préciser `--name-status` pour voir la liste des fichiers commités :
+
+```sh
+$ git log --name-status -- **/*.js
+```
+
+### Trouver un tag où un commit est référencé
+
+Pour trouver tous les tags contenant un commit en particulier :
+
+```sh
+$ git tag --contains <idducommit>
+```
+
+## Sous-modules
+
+<a name="clone-submodules"></a>
+### Cloner tous les sous-modules
+
+```sh
+$ git clone --recursive git://github.com/foo/bar.git
+```
+
+Si ces derniers ont déjà été clonés :
+
+```sh
+$ git submodule update --init --recursive
+```
+
+<a name="delete-submodule"></a>
+### Retirer un sous-module
+
+Créer un sous-module est assez trivial, mais le supprimer l'est moins. Les commandes dont vous avez besoin sont :
+
+```sh
+$ git submodule deinit nomdusubmodule
+$ git rm nomdusubmodule
+$ git rm --cached nomdusubmodule
+$ rm -rf .git/modules/nomdusubmodule
+```
+
+## Objets divers
+
+### Récupérer un fichier supprimé
+
+Trouvez le commit le plus récent où le fichier existait encore :
+
+```sh
+$ git rev-list -n 1 HEAD -- nomdufichier
+```
+
+Puis faites un `checkout` sur ce fichier :
+
+```
+git checkout idducommitquiasupprimé^ -- nomdufichier
+```
+
+### Supprimer un tag
+
+```sh
+$ git tag -d <nom_du_tag>
+$ git push <remote> :refs/tags/<nom_du_tag>
+```
+
+<a name="recover-tag"></a>
+### Récupérer un tag supprimé
+
+Si vous voulez récupérer un tag qui a déjà été supprimé, vous pouvez le faire en suivant ces étapes : Tout d'abord, vous devez retrouver le tag inaccessible en question :
+
+```sh
+$ git fsck --unreachable | grep tag
+```
+
+Notez le hash du tag dans un coin. Puis, restaurez le tag supprimé avec la commande suivante, qui utilise [`git update-ref`](https://git-scm.com/docs/git-update-ref) :
+
+```sh
+$ git update-ref refs/tags/<nom_du_tag> <hash>
+```
+
+Votre tag devrait maintenant être restauré.
+
+### Patch supprimé
+
+Si quelqu'un vous a envoyé une Pull Request sur Github, mais a par la suite supprimé son fork originel, vous ne serez pas en mesure de cloner son dépôt ou d'utiliser `git am` étant donné que les URLs de [.diff, .patch](https://github.com/blog/967-github-secrets) sont devenues indisponibles. Mais vous pouvez vous déplacer sur la PR en elle-même en utilisant les [les refs spéciaux de GitHub](https://gist.github.com/piscisaureus/3342247). Pour récupérer le contenu de PR#1 vers une nouvelle branche intitulée pr_1 :
```

Comment:
```suggestion
Si quelqu'un vous a envoyé une Pull Request sur GitHub, mais a par la suite supprimé son fork originel, vous ne serez pas en mesure de cloner son dépôt ou d'utiliser `git am` étant donné que les URLs de [.diff, .patch](https://github.com/blog/967-github-secrets) sont devenues indisponibles. Mais vous pouvez vous déplacer sur la PR en elle-même en utilisant les [les refs spéciaux de GitHub](https://gist.github.com/piscisaureus/3342247). Pour récupérer le contenu de PR#1 vers une nouvelle branche intitulée pr_1 :
```
"Github" -> "GitHub" (I think I mistyped that too in a comment earlier, but it's just a comment, so whatever 😂